app/geo/gui/src/ts/twxplore/gui/geo/attributeStrategies/attributeStrategies.ts

```diff
@@ -0,0 +1,82 @@
+// In an OOP Language -
+// TypeScript
+interface FeatureAttributeFunctions {
+    isNumeric() : boolean
+    isString(): boolean
+}
+
+enum Attributes {
+    LABEL = "label",
+    TYPE = "type",
+    URI = "uri",
+    FREQUENCY = "frequency",
+    TIMESTAMP = "timestamp",
+    LOCALITY = "locality",
+    POSTALCODE = "postalcode",
+    REGIONS =  "regions",
+    TRANSMISSIONPOWER = "transmissionPower",
+    GEOMETRY = "geometry",
+}
+
+export class FeatureAttribute implements FeatureAttributeFunctions{
+    private attributeName: string 
+    private subclass: any
+    constructor(attributeName: string) {
+        this.attributeName = attributeName
+        this.subclass = null
+        switch (this.attributeName) {
+            case (Attributes.TIMESTAMP): {
+                this.subclass = FeatureAttributeTimestamp.getInstance()
```

Comment:
Not sure what's going on here. You need a separate static method that takes a name (string or the enum above) and returns a FeatureAttribute.

static valueOf(name, ... other parameters ...) { return an instance of the appropriate subclass } 

It's a factory method.